[DOC-588][docs] Voyager docs for v1.8.8 (#25415)
yugabyte/yugabyte-db: docs/content/preview/yugabyte-voyager/release-notes.md

What follows are the release notes for the YugabyteDB Voyager v1 release series. Content will be added as new notable features and changes are available in the patch releases of the YugabyteDB v1 series.

## v1.8.8 - December 24, 2024

### Enhancements

- Assessment and Schema Analysis Reports

>   - [assess-migration](../reference/assess-migration/) and [analyze-schema](../reference/schema-migration/analyze-schema/) now accept a new flag `--target-db-version`, which allows the user to specify the version of YugabyteDB they wish to assess/analyze against. It defaults to the latest stable release version (currently 2024.2.0.0).

Author: Do we have a short code for the current release? @ddhodge

>   - Presence of advisory locks, XML functions, and system columns in DDLs will now be detected and reported in assessment and schema analysis reports.

Reviewer: ```suggestion
  - Assessment and schema analysis now detect and report the presence of advisory locks, XML functions, and system columns in DDLs.
```

>   - Presence of [large objects](https://www.postgresql.org/docs/current/largeobjects.html) (and its functions) in DDLs/DMLs is now detected and reported in assessment and schema analysis reports.

Reviewer: ```suggestion
  - Assessment and schema analysis now detect the presence of [large objects](https://www.postgresql.org/docs/current/largeobjects.html) (and their functions) in DDLs/DMLs.
```

>   - In the Schema analysis report (html/text), a few field names have been modified to improve readability. Invalid Count -> Objects with Issues, Total Count -> Total Objects, Valid Count -> Objects without Issues. The logic of determining when an object is considered to have issues or not has also been improved.

Reviewer: ```suggestion
  - In the Schema analysis report (html/text), changed the following field names to improve readability: Invalid Count to Objects with Issues; Total Count to Total Objects; and Valid Count to Objects without Issues. The logic determining when an object is considered to have issues or not has also been improved.
```

>   - Stop reporting Unlogged tables as an issue in assessment and schema analysis reports by default, as it has been fixed in YugabyteDB {{<release "2024.2.0.0">}}.

Reviewer: "it" is https://docs.yugabyte.com/preview/yugabyte-voyager/known-issues/postgresql/#unlogged-table-is-not-supported so maybe link to it
"Stop reporting unlogged tables as an issue in assessment and schema analysis reports by default, as [the issue](link) has been fixed in YugabyteDB {{<release "2024.2.0.0">}}."
Is unlogged really supported in 2024.2? https://docs.yugabyte.com/stable/api/ysql/the-sql-language/statements/ddl_create_table/#unlogged

Reviewer 2: Yeah, the subtlety is that it's not "fixed" as such, just ignored... But from voyager's perspective we decided not to report it as an "issue" anymore because the user doesn't really have to manually intervene. 

We can perhaps phrase it as "... as the UNLOGGED option no longer results in a syntax error in YugabyteDB 2024.2.0.0" ?

Author: I've rephrase id as @makalaaneesh said.

>   - Stop reporting [ALTER PARTITIONED TABLE ADD PRIMARY KEY](https://github.com/yugabyte/yb-voyager/issues/612) as an issue by default in assessment and schema analysis reports, as it has been fixed in YugabyteDB.

Reviewer: - Stop reporting [ALTER PARTITIONED TABLE ADD PRIMARY KEY](https://github.com/yugabyte/yb-voyager/issues/612) as an issue in assessment and schema analysis reports by default, as [the issue](link to https://docs.yugabyte.com/preview/yugabyte-voyager/known-issues/postgresql/#adding-primary-key-to-a-partitioned-table-results-in-an-error) has been fixed in YugabyteDB [release].

Author: @makalaaneesh, could you please clarify the release version here?

Reviewer 2: It's technically 2024.1.0.0, 2024.2.0.0, and 2.23.0.0. 
But as far as release notes go, I think we can just mention "... has been fixed in YugabyteDB from 2024.1.0.0 onwards" ?

  - In the assessment report, only statements from `pg_stat_statements` that belong to the schemas provided by the user will be processed for detecting and reporting issues.
- Data Migration

>   - `import data file` and `import data to source replica` now accept a new flag `truncate-tables` (in addition to `import data`), which, when used with `start-clean true`, truncates all the tables in the target/source-replica DB before importing data into the tables.

Reviewer: ```suggestion
  - `import data file` and `import data to source replica` now accept a new flag `truncate-tables` (in addition to `import data`), which, when used with `start-clean true`, truncates all the tables in the target/source-replica database before importing data into the tables.
```
